model.go

```diff
@@ -16,11 +16,17 @@ type ModelInfo struct {
 	Provider    string `json:"provider"`
 	Host        string `json:"host"`
 	Path        string `json:"path"`
+	Dedicated   bool   `json:"dedicated"`
 }
 
 func (m ModelInfo) Hash() string {
 	// Concatenate the attributes into a single string
 	data := fmt.Sprintf("%v|%v|%v|%v", m.Name, m.SourceModel, m.Provider, m.Host)
+	// Don't include the "dedicated" attribute if host is NOT "hypermode" or
+	// if it's NOT "dedicated" (default)
+	if m.Host == "hypermode" && m.Dedicated {
+		data += fmt.Sprintf("|%v", m.Dedicated)
+	}
```

Comment:
note: the plan is to migrate existing Hypermode-hosted models to shared models as per the notion doc. we decided this breaking change is okay, as it would also be a good test to see if any of our current users "care" about dedicated models or not.

so once this goes live, models who don't have this "dedicated" field set, will default to false, and so the hash won't change.

and if they decide to add `"dedicated": true`, then the hash will change and we can do the necessary steps to deploy it as a dedicated model. same goes if they revert it back to false, the hash will change again.